### DISCUSSION TITLE
[@types/xml-crypto] - FileKeyInfo getKey method accepts wrong parameter

### DISCUSSION THREAD

**Discussion by djaqua** [DefinitelyTyped/DefinitelyTyped#63932] [Issues with a @types package]
Firstly, I'd like to express my gratitude that someone has taken the time to provide type definitions for this package! That said, there seems to be a problem with the method signature for `FileKeyInfo::getKey`. The problem is that getKey currently  accepts exactly one KeyInfo node, but the underlying implementation passes an array of KeyInfo nodes. 

`SignedXml.prototype.loadSignature` is the only place that initializes `this.keyInfo` to a non-null value :
```
this.keyInfo = xpath.select(".//*[local-name(.)='KeyInfo']", signatureNode); // SELECT RETURNS AN ARRAY
```

`SignedXml.prototype.checkSignature` is the only place that uses `FileKeyInfo::getKey`: 
```
this.signingKey = this.keyInfoProvider.getKey(this.keyInfo); // this passes an array to getKey
```

In a properly constructed Signature element, there would only be one KeyInfo element in the array, but it is an array nonetheless. I propose modifying the method signature, changing it from `getKey(keyInfo?: Node): Buffer` to `getKey(keyInfo?: Node | Node[]): Buffer`

Without that modification, clients of @types/xml-crypto must cast keyInfo as a Node array in order to process the KeyInfo element correctly. If I'm being stupid, lets have a conversation :smiling_face_with_tear:

[a comment by an automated account is not shown]

**Comment by eheikes** [comment]
Makes sense to me @djaqua. Want to create a PR?

I don't think the `keyInfo` can even be a `Node`, can it? As you say, the `xpath.select()` will always return an array, and it's initialized as `null`. So maybe `getKey(keyInfo?: null | Node[]): Buffer`?

**Reply by djaqua** [reply]
I'm definitely open to that suggestion. It turns out that I've now become one of the maintainers of `xml-crypto` and we're working on all the bits that affect these type definitions, so if you don't mind, I guess lets table this for now until those changes have been made and then see how the dust settles?

**Reply by eheikes** [reply]
That's fine by me.
